virtcontainers/sandbox.go

```diff
@@ -561,6 +566,23 @@ func (s *Sandbox) Status() SandboxStatus {
 	}
 }
 
+// Monitor returns a error channel for watcher to watch at
+func (s *Sandbox) Monitor() (chan error, error) {
+	if s.state.State != StateRunning {
+		return nil, fmt.Errorf("Sandbox not running")
+	}
+
+	if s.monitor == nil {
+		s.Lock()
+		if s.monitor == nil {
+			s.monitor = newMonitor(s)
+		}
+		s.Unlock()
+	}
+
+	return s.monitor.newWatcher()
```

Comment:
I am a bit concerned about the way this is handled here. The caller can basically call a million times into `Monitor()`, this will result into a million go routines being spawned here. If think we should try to handle things better, by returning the existing channel from `newWatcher()` if it already exists. This means we would always have only one channel, because I don't see the usage for having several channels referring to the same thing. Maybe I am missing something.

Reply:
No, the monitor makes sure there is only one watcher goroutine per monitor. We just add more watch channels to it.

Reply:
Oh I see. So we'll have one go routine with an infinite number of channels. Sounds good.